infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/TopologyDescriptor.java

```diff
@@ -30,14 +30,30 @@ public class TopologyDescriptor {
     @Getter
     private final long topologyConfigId;
 
+    // Contains remote clusters that are SOURCE to the local cluster.
     @Getter
-    private final Map<String, ClusterDescriptor> sourceClusters;
+    private final Map<String, ClusterDescriptor> remoteSourceClusters;
 
+    // Contains remote clusters that are SINK to the local cluster.
     @Getter
-    private final Map<String, ClusterDescriptor> sinkClusters;
+    private final Map<String, ClusterDescriptor> remoteSinkClusters;
 
+    // Map of remote ClusterDescriptor -> ReplicationModels. Contains clusters which will be SOURCE (w.r.t local cluster)
+    // and the corresponding replication models
     @Getter
-    private final Map<String, ClusterDescriptor> invalidClusters;
+    private final Map<ClusterDescriptor, Set<LogReplication.ReplicationModel>> remoteSourceClusterToReplicationModels;
+
+    // Map of remote ClusterDescriptor -> ReplicationModels. Contains clusters which will be SINK (w.r.t local cluster)
+    // and the corresponding replication models
+    @Getter
+    private final Map<ClusterDescriptor, Set<LogReplication.ReplicationModel>> remoteSinkClusterToReplicationModels;
```

Comment:
SessionManager also tracks `incomingSessions` and `outgoingSessions` just wondering if it might help to just have it hang on to the TopologyDescriptor to reduce the number of maps tracking the sessions perhaps? Do you think it can help reduce the overhead in keeping these in sync?

Reply:
I think we'll need these constructs separately..the reason is that for full_table, incomingSessions is always the SINK and outgoingSession is always the SOURCE. 
But with more variable, replicationModels and who is the connection initiators and when/how sessions are created, having them separate makes it easier. Example, for Logical_group use case we will creating sessions only when we know the destinations.

Reply:
Will add a TODO for myself, so I can evaluate in the connectionModel PR.